src/unix.rs

```diff
@@ -102,10 +102,18 @@ fn localized_msg_fmt(msg: &[u8], f: &mut Formatter) -> fmt::Result {
 }
 
 pub fn errno_fmt(e: i32, f: &mut Formatter) -> fmt::Result {
-    let msg = unsafe {
-        let msg = strerror(e) as *const c_char;
-        slice::from_raw_parts(msg as *const u8, strlen(msg))
-    };
+    // 128 bytes should be long enough for all error messages
+    const BUF_SIZE: usize = 128;
+
+    // The cast is neessary for portability, since c_char is not necessarily i8.
```

Comment:
c_char is necessary 8-bit according to POSIX, and thus it has same representation as `i8` or/and `u8`. No need to use c_char

Reply:
The Clippy lint suggests to use `0_i8` instead, which doesn’t work because as stated `c_char` is not necessarily `i8`. That said, a type annotation would also work here so I’ve changed it to that.